### DISCUSSION TITLE
Modifying class member variables in other mods.

### DISCUSSION THREAD

**Discussion by Desynq** [FabricMC/fabric-api#2350] [Mod Dev Support] [answered]
I'm wondering on how I can modify a member variable within another mod's class file.

For instance, I'm trying to modify the window width of the Origin mod's display screen.

I've narrowed down where this occurs to io.github.apace100.origins.screen.OriginDisplayScreen at line 31.
It's declared as a protected static final integer (protected static final int windowWidth = 176;).

While I know that mixin can inject and modify variables within methods, I'm wondering how I can accomplish this with variables declared out of the scope of a method but within a class.

Any help on this subject is greatly appreciated, thank you.

**Comment by Desynq** [comment]
Nevermind, I cloned the mod and tried out changing the windowWidth to double its current amount (from 176 to 352) and it broke a lot of things. I think I'm fine without a proper answer to this.

**Comment by warjort** [comment, marked as answer]
In case you want to know the real answer for future reference,
you should use an accessor
https://fabricmc.net/wiki/tutorial:mixin_accessors
which you will need to annotate as ````@Mutable```` to workaround the "final".

This is really just a quick way of writing a normal mixin with a shadowed mutable field and then adding your own interface to access the field.

However there is a gotcha. There is no guarantee the code using that final field actually accesses it at runtime.
The compiler is allowed to "inline" the values of static final fields into referencing methods at compile time.
So changing the value of the static final field will do nothing if that happens.

**Reply by Desynq** [reply]
Could you possibly give an example of how I would do it in this instance, I'm kind of confused by the tutorial's examples, hope you don't mind.

**Reply by warjort** [reply]
Something like (untested, not even compiled):
````
@Mixin(OriginDisplayScreen.class)
public interface OriginDisplayScreenAccessor {
  @Accessor("windowWidth")
  @Mutable
  public static void setWindowWidth(int windowWidth) {
    throw new AssertionError();
  }
}

// Use it
OriginDisplayScreenAccessor.setWindowWidth(200);
````

**Reply by Desynq** [reply]
ok, thank you.
